src/game/shared/neo/neo_player_shared.cpp

```diff
@@ -110,6 +110,66 @@ bool ClientWantsAimHold(const CNEO_Player* player)
 #endif
 }
 
+#ifdef CLIENT_DLL
+extern ConVar cl_neo_player_pings;
+#endif // CLIENT_DLL
+void CheckPingButton(CNEO_Player* player)
+{
+	if (!player->IsAlive() || !(player->m_afButtonPressed & IN_ATTACK3) || player->m_flNextPingTime > gpGlobals->curtime)
+	{
+		return;
+	}
+
+	if (!player->IsBot())
+	{ // players with pings disabled can't ping
+#ifdef GAME_DLL
+		const bool showPlayerPings = atoi(engine->GetClientConVarValue(engine->IndexOfEdict(player->edict()), "cl_neo_player_pings"));
+#else
+		const bool showPlayerPings = cl_neo_player_pings.GetBool();
+#endif // GAME_DLL
+		if (!showPlayerPings)
+		{
+			return;
+		}
+	}
+
+	IGameEvent* event = gameeventmanager->CreateEvent("player_ping");
+	if (event)
+	{
+		trace_t tr;
+		Vector forward;
+		player->EyeVectors(&forward);
+		Vector eyePosition = player->EyePosition();
+		UTIL_TraceLine(eyePosition, eyePosition + forward * MAX_COORD_RANGE, MASK_VISIBLE_AND_NPCS, player, COLLISION_GROUP_NONE, &tr);
+
+		if (Q_stristr(tr.surface.name, "SKYBOX"))
+		{
+			return;
+		}
+
+		event->SetInt("userid", player->GetUserID());
+		event->SetInt("playerTeam", player->GetTeamNumber());
```

Comment:
We're setting `playerTeam` here, but reading `playerteam` (different capitalization) in the `CNEOHud_PlayerPing::FireGameEvent`. Since the event code is owned by the engine and we don't have control over the implementation, maybe it'd be safer to unify the capitalization here, so that we can't spontaneously break if Valve ships an update to the engine logic that handles this.